src/query/atomWithQuery.ts

```diff
@@ -28,76 +28,60 @@ export function atomWithQuery<
       ) => AtomQueryOptions<TQueryFnData, TError, TData, TQueryData>),
   equalityFn: (a: TData, b: TData) => boolean = Object.is
 ): WritableAtom<TData, Action> {
-  const stateAtom = atom(() => {
-    const state = {
-      resolve: null as ((data: TData) => void) | null,
-      setData: null as ((data: TData) => void) | null,
-      prevData: null as TData | null,
-      unsubscribe: null as (() => void) | null,
-      handle(result: QueryObserverResult<TData, TError>) {
+  const queryDataAtom = atom(
+    (get) => {
+      const queryClient = get(queryClientAtom)
+      const options =
+        typeof createQuery === 'function' ? createQuery(get) : createQuery
+      let resolve: ((data: TData) => void) | null = null
+      const dataAtom = atom<TData | Promise<TData>>(
+        new Promise<TData>((r) => {
+          resolve = r
+        })
+      )
+      let setData: (data: TData) => void = () => {
+        throw new Error('setting data without mount')
```

Comment:
```suggestion
        throw new Error('atomWithQuery: setting data without mount')
```

Reply:
I think the more intention we put into these, the more helpful it will be for users.

When is it likely this error will happen?

Reply:
It's meant for detecting bugs, not really for users.

Hm, it could happen when you refetch before mount. So, users would actually sees it if they invoke `refetch` in render. (we could avoid the error by setting `setData` before refetch, but that would only cause infinite loop. 🤔 Maybe the suggestion is an easy fix.)